processing/src/main/java/org/apache/carbondata/processing/csvreaderstep/UnivocityCsvParser.java

```diff
@@ -41,6 +41,10 @@
 public class UnivocityCsvParser {
 
   /**
+   * Max number of columns that will be parsed for a row by univocity parsing
+   */
+  private static final int MAX_NUMBER_OF_COLUMNS_FOR_PARSING = 2000;
```

Comment:
If array creation is for every row, then we should surely control it.
If array creation is one time, then can give value of 2000 as default. and can add DataLoad command option to control it, mainly useful for wide tables